incydr/cli/cmds/watchlists.py

```diff
@@ -0,0 +1,583 @@
+
+MAX_USER_DISPLAY_COUNT = 25
+
+
+def get_watchlist_id_callback(ctx, param, value):
+    if not value:
+        return
+    try:
+        UUID(hex=value)
+        return value
+    except ValueError:
+        # if not an ID value
+        client = ctx.obj()
+        return client.watchlists.v1.get_id_by_name(value)
+
+
+@click.group(cls=IncydrGroup)
+@log_level_option
+@log_file_option
+@click.pass_context
+def watchlists(ctx, log_level, log_file):
+    """
+    View and manage watchlists.
+
+    After creation, Watchlists can be managed by type (ex: `DEPARTING_EMPLOYEE`) or ID. `CUSTOM` watchlists must be managed by title or ID.
+
+    The following values are valid watchlist types:
+
+        * CONTRACT_EMPLOYEE
+        * DEPARTING_EMPLOYEE
+        * ELEVATED_ACCESS_PRIVILEGES
+        * FLIGHT_RISK
+        * HIGH_IMPACT_EMPLOYEE
+        * NEW_EMPLOYEE
+        * PERFORMANCE_CONCERNS
+        * POOR_SECURITY_PRACTICES
+        * SUSPICIOUS_SYSTEM_ACTIVITY
+        * CUSTOM
+    """
+    init_client(ctx, log_level, log_file)
+
+
+watchlist_arg = click.argument("watchlist", callback=get_watchlist_id_callback)
+csv_option = click.option(
+    "--csv", is_flag=True, default=False, help="Users are specified in a CSV file. "
+)
+
+
+@watchlists.command("list", cls=IncydrCommand)
+@click.option(
+    "--user",
+    default=None,
+    help="Filter by watchlists where the user is a member.  Accepts a user ID or a username.  Performs an additional lookup if a username is passed",
+    callback=user_lookup_callback,
+)
+@table_format_option
+@columns_option
+@click.pass_context
+def list_(
+    ctx: Context, user: str = None, format_: TableFormat = None, columns: str = None
+):
+    """
+    List watchlists.
+    """
+    client = ctx.obj()
+    watchlists = client.watchlists.v1.iter_all(user_id=user)
+    _output_results(watchlists, Watchlist, format_, columns)
+
+
+@watchlists.command(cls=IncydrCommand)
+@watchlist_arg
+@click.pass_context
+def show(ctx: Context, watchlist: str = None):
+    """
+    Show details for a watchlist.
+
+    WATCHLIST can be specified by watchlist type (ex: `DEPARTING_EMPLOYEE`) or ID.
+    `CUSTOM` watchlists must be specified by title or ID.
+
+    If using `rich`, outputs a summary of watchlist information and membership. This includes the following:
+
+    * included_users
+    * excluded_users
+    * included_departments
+    * included_directory_groups
+
+    Lists of users will be truncated to only display the first 25 members, use the `list-included-users` and `list-excluded-users`
+    commands respectively to see more details.
+
+    If not using `rich`, outputs watchlist information in JSON without additional membership summary information.
+    """
+    client = ctx.obj()
+    watchlist_response = client.watchlists.v1.get(watchlist)
+
+    if not client.settings.use_rich:
+        console.print(watchlist_response.json(), highlight=False)
+
+    def models_as_table(model, models, title=None):
```

Comment:
This would probably be useful to stick in `incydr.utils`.

Reply:
Would it be more fitting in `render` sits its specific to rich formatting